File: java/FindUniqueNumberInAnArrayWhereAllNumbersArePresentTwice.java
class Algorithms { 
	static int ans(int arr[], int n) {  
		int sum = arr[0]; 

		for (int i=1;i<n;i++) 
			sum = sum ^ arr[i]; 
	
		return sum; 
	} 

	public static void main (String[] args) { 
		int arr[] = {1,2,4,5,2,5,4,1,8};
		System.out.println(ans(arr, arr.length)); 
	} 
} 

File: python/FindUniqueNumberInAnArrayWhereAllNumbersArePresentTwice.py
def ans(arr,n): 
	
	sum = arr[0] 
	for i in range(1,n): 
		sum = sum^arr[i] 
	
	return sum 

arr = [1,2,4,5,2,5,4,1,8] 
print (ans(arr, len(arr))) 

